pkgs/development/libraries/libhugetlbfs/default.nix

```diff
@@ -0,0 +1,56 @@
+{ stdenv, lib, fetchurl, fetchpatch }:
+
+stdenv.mkDerivation rec {
+  pname = "libhugetlbfs";
+  version = "2.22";
+
+  src = fetchurl {
+    url = "https://github.com/libhugetlbfs/libhugetlbfs/releases/download/${version}/libhugetlbfs-${version}.tar.gz";
+    sha256 = "11b7k8xvgx68rjzidm12a6l6b23hwi7hj149y9xxfz2j5kmakp4l";
+  };
+
+  outputs = [ "bin" "dev" "man" "doc" "lib" "out" ];
+
+  patches = [
+    # Don't check that 32-bit and 64-bit libraries don't get installed
+    # to the same place if only one platform is being built for.
+    # Can be removed if build succeeds without it.
+    (fetchpatch {
+      url = "https://groups.google.com/forum/message/raw?msg=libhugetlbfs/IswjDAygfwA/PKy7MZbVAAAJ";
+      sha256 = "00fyrhn380d6swil8pcf4x0krl1113ghswrvjn3m9czc3h4p385a";
+    })
+  ];
+
+  postConfigure = ''
+    patchShebangs ld.hugetlbfs
+  '';
+
+  enableParallelBuilding = true;
+  makeFlags = [
+    "BUILDTYPE=NATIVEONLY"
+    "PREFIX=$(out)"
+    "HEADERDIR=$(dev)/include"
+    "LIBDIR32=$(lib)/$(LIB32)"
+    "LIBDIR64=$(lib)/$(LIB64)"
+    "EXEDIR=$(bin)/bin"
+    "DOCDIR=$(doc)/share/doc/libhugetlbfs"
+  ] ++ map (n: "MANDIR${n}=$(man)/share/man/man${n}")
+    (lib.genList (n: toString (n + 1)) 8);
+
+  buildPhase = ''
+    # Default target builds tests as well, and the tests want a static
+    # libc.
+    make libs tools $makeFlags
+  '';
+
+  installPhase = ''
+    make install install-docs $makeFlags
```

Comment:
When overriding `installPhase`, you should also provide `runHook preInstall` and `runHook postInstall`. Though, in this case, why not rely on the default `installPhase` and specify `installTargets = [ "install" "install-docs" ];`?

Reply:
> When overriding `installPhase`, you should also provide `runHook preInstall` and `runHook postInstall`.

Why? As I understand it, the hooks are stdenv hooks, not derivation hooks, and are for the derivation to use if it wants to. It’s not like the hooks are a generic, consistent way to allow users of packages to insert extra commands, because if I had used `preInstall` here, somebody overriding that would break the package.

> Though, in this case, why not rely on the default `installPhase` and specify `installTargets = [ "install" "install-docs" ];`?

Because I didn’t know about those! Thanks!

Reply:
You would use `pkg.overrideAttrs (attrs: {postInstall = (attrs.postInstall or "") + ''foo'';})`. I keep recommending adding hooks because I have seen their lack of confuse nixpkgs users several times. Granted, nine times out of ten it was about `fixupPhase` hooks, but `installPhase` was there too. I think consistency is useful for reducing the amount of head scratchers.

Reply:
Fair enough — thanks for explaining!